libtransmission/rpc-server.cc

```diff
@@ -1084,8 +1084,20 @@ tr_rpc_server::tr_rpc_server(tr_session* session_in, tr_variant* settings)
     }
 
     key = TR_KEY_rpc_socket_mode;
+    bool is_missing_rpc_socket_mode_key = true;
 
-    if (!tr_variantDictFindInt(settings, key, &i))
+    if (tr_variantDictFindStrView(settings, key, &sv))
+    {
+        /* Read the socket permission as a string representing an octal number. */
+        is_missing_rpc_socket_mode_key = false;
+        std::from_chars(sv.data(), sv.data() + sv.size(), i, 8);
```

Comment:
tr_parseNum() will be more portable here; it resolves to from_chars on platforms that have it.

Reply:
Ok, I took a look at `tr_parseNum()` and since the base of both implementations already supports dealing with radices, I decided to rather add an optional radix parameter to `tr_parseNum()` than to use it as-is and then convert the value from decimal to octal manually.

It works fine for all the common bases (8, 10, 16), but if a different radix is specified, the behaviour of the `stringstream` and `from_chars` diverges, because `from_chars` handles any base from 2 to 36 while the `stringstream` only handles the values 8, 10 and 16, every other value leads to prefix dependant parsing which in our case would result in decimal. This should be kept in mind but I do not think that it would require additional checks, because this parametrization is not user-facing in any way.
I will add an additional comment stating this behaviour.